src/components/Menu/Menu.jsx

```diff
@@ -0,0 +1,147 @@
+import React, {useCallback, useState, useEffect} from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'clsx';
+import styles from './Menu.scss';
+
+export const Menu = ({children, isDisplayed, minWidth, maxWidth, className, onMouseEnter, onMouseLeave, anchorEl, anchorPosition, onClose, ...props}) => {
+    const [stylePosition, setStylePosition] = useState();
+
+    useEffect(() => {
+        if (isDisplayed) {
+            setPositioningStyles(anchorEl);
+        }
+    }, [anchorEl, isDisplayed, setPositioningStyles]);
+
+    const setPositioningStyles = useCallback(() => {
+        const positioning = getPositioningStyle(anchorEl);
+        const positioningStyles = {
+            top: `calc(${positioning.top}px + ${anchorPosition.top})`,
+            left: `calc(${positioning.left}px + ${anchorPosition.left})`
+        };
+
+        setStylePosition(positioningStyles);
+    }, [getPositioningStyle, anchorEl, anchorPosition]);
+
+    const getPositioningStyle = useCallback(() => {
+        if (!anchorEl) {
+            return {
+                top: 0,
+                left: 0
+            };
+        }
+
+        let el = anchorEl.getBoundingClientRect();
+        return {
+            top: el.top + el.height,
+            left: el.left
+        };
+    }, [anchorEl]);
+
+    // ---
+    // Styling
+    // ---
+    const styleMenu = {...stylePosition};
+    if (minWidth) {
+        styleMenu.minWidth = minWidth;
+    }
+
+    if (maxWidth) {
+        styleMenu.maxWidth = maxWidth;
+    }
+
+    // ---
+    // Render
+    // ---
+    if (isDisplayed) {
+        return (
+            <div className={classnames(styles.menu_wrapper)}>
```

Comment:
We will need the backdrop to be optional to be able to use the menu as a submenu (only the root menu has a backdrop)